File: backend/app/api/admin_api.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from pydantic import BaseModel
from typing import List

from app.db.session import SessionLocal
from app.models.user import User
from app.api.deps import get_db, get_current_user

router = APIRouter()

class UnverifiedUserResponse(BaseModel):
    id: int
    email: str
    full_name: str
    role: str

    class Config:
        from_attributes = True

def require_admin(current_user: User = Depends(get_current_user)):
    if current_user.email != "admin@ksp.gov.in":
        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Admin privileges required")
    return current_user

@router.get("/unverified-users", response_model=List[UnverifiedUserResponse])
def get_unverified_users(db: Session = Depends(get_db), admin_user: User = Depends(require_admin)):
    users = db.query(User).filter(User.is_verified_by_admin == False, User.email != "admin@ksp.gov.in", User.status != "rejected").all()
    return users

@router.get("/verified-users", response_model=List[UnverifiedUserResponse])
def get_verified_users(db: Session = Depends(get_db), admin_user: User = Depends(require_admin)):
    users = db.query(User).filter(User.is_verified_by_admin == True, User.email != "admin@ksp.gov.in", User.status != "removed").all()
    return users

@router.post("/verify-user/{user_id}")
def verify_user(user_id: int, db: Session = Depends(get_db), admin_user: User = Depends(require_admin)):
    user = db.query(User).filter(User.id == user_id).first()
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    
    user.is_verified_by_admin = True
    db.commit()
    
    return {"message": "User verified successfully"}

@router.post("/reject-user/{user_id}")
def reject_user(user_id: int, db: Session = Depends(get_db), admin_user: User = Depends(require_admin)):
    user = db.query(User).filter(User.id == user_id).first()
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    
    user.status = "rejected"
    db.commit()
    
    return {"message": "User rejected successfully"}

@router.post("/remove-user/{user_id}")
def remove_user(user_id: int, db: Session = Depends(get_db), admin_user: User = Depends(require_admin)):
    user = db.query(User).filter(User.id == user_id).first()
    if not user:
        raise HTTPException(status_code=404, detail="User not found")
    
    db.delete(user)
    db.commit()
    
    return {"message": "User removed successfully"}

File: frontend/src/lib/adminApi.ts
export interface UnverifiedUser {
  id: number;
  email: string;
  full_name: string;
  role: string;
}

export const getUnverifiedUsers = async (token: string): Promise<UnverifiedUser[]> => {
  const response = await fetch('https://crimevision-aq07.onrender.com/api/admin/unverified-users', {
    headers: {
      'Authorization': `Bearer ${token}`
    }
  });
  if (!response.ok) throw new Error('Failed to fetch unverified users');
  return response.json();
};

export const verifyUserByAdmin = async (userId: number, token: string) => {
  const response = await fetch(`https://crimevision-aq07.onrender.com/api/admin/verify-user/${userId}`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${token}`
    }
  });
  if (!response.ok) throw new Error('Failed to verify user');
  return response.json();
};

export const rejectUserByAdmin = async (userId: number, token: string) => {
  const response = await fetch(`https://crimevision-aq07.onrender.com/api/admin/reject-user/${userId}`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${token}`
    }
  });
  if (!response.ok) throw new Error('Failed to reject user');
  return response.json();
};

export const getVerifiedUsers = async (token: string): Promise<UnverifiedUser[]> => {
  const response = await fetch('https://crimevision-aq07.onrender.com/api/admin/verified-users', {
    headers: {
      'Authorization': `Bearer ${token}`
    }
  });
  if (!response.ok) throw new Error('Failed to fetch verified users');
  return response.json();
};

export const removeUserByAdmin = async (userId: number, token: string) => {
  const response = await fetch(`https://crimevision-aq07.onrender.com/api/admin/remove-user/${userId}`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${token}`
    }
  });
  if (!response.ok) throw new Error('Failed to remove user');
  return response.json();
};
